guildwars2/guild/general.py

```diff
@@ -213,9 +248,50 @@ class GeneralGuild:
                         inline=False)
                     counter += 1
         if counter == 0:
-            return await ctx.send(
-                "No stash log entries yet for {}".format(guild_name.title()))
+            return await ctx.send("No stash log entries yet for {}".format(
+                guild_name.title()))
         try:
             await ctx.send(embed=data)
         except discord.Forbidden:
             await ctx.send("Need permission to embed links")
+
+    @guild.command(name="default", usage="<guild name>")
+    @commands.guild_only()
+    @commands.cooldown(1, 10, BucketType.user)
+    async def guild_default(self, ctx, *, guild_name=None):
+        """ Set your preferred guild for guild commands on this Discord Server.
+        Commands from the guild command group invoked
+        without a guild name will default to this guild.
+
+        Invoke this command without an argument to reset the default guild.
+        """
+        guild = ctx.guild
+        if guild_name is None:
+            await self.bot.database.set_guild(guild, {
+                "guild_ingame": None,
+            }, self)
+            await ctx.send("Your preferred guild is now reset for this server."
+                           )
+        else:
+            endpoint_id = "guild/search?name=" + guild_name.replace(' ', '%20')
+            # Guild ID to Guild Name
+            try:
+                guild_id = await self.call_api(endpoint_id)
+                guild_id = guild_id[0]
+            except (IndexError, APINotFound):
+                return await ctx.send("Invalid guild name")
+            except APIForbidden:
+                return await ctx.send(
+                    "You don't have enough permissions in game to "
+                    "use this command")
+            except APIError as e:
+                return await self.error_handler(ctx, e)
+
+            # Write to DB, overwrites existing guild
+            await self.bot.database.set_guild(guild, {
+                "guild_ingame": guild_id,
+            }, self)
+
+            await ctx.send(
+                "Your preferred guild is now set to {0} for this server"
+                .format(guild_name.title()))
```

Comment:
Should include a more helpful for message for the user eg:
```py
"Your default guild is now set to {} for this server. "
"All commands from the `guild` command group "
"invoked without a specified guild will default to "
"this guild. To reset, simply invoke this command "
"without specifying a guild"
```

Reply:
Changed